src/connectedk8s/azext_connectedk8s/_client_factory.py

```diff
@@ -2,16 +2,31 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
+
 from azure.cli.core.commands.client_factory import get_mgmt_service_client
 from azure.cli.core.profiles import ResourceType
 from azure.cli.core._profile import Profile
+from azure.cli.core import telemetry
+from azure.cli.core.azclierror import ValidationError
 from azure.cli.core.commands.client_factory import configure_common_settings
 from azure.cli.core.commands.client_factory import get_subscription_id
 from azure.graphrbac import GraphRbacManagementClient
 
+import os
+import requests
+import azext_connectedk8s._constants as consts
+from collections import namedtuple
+
+AccessToken = namedtuple("AccessToken", ["token", "expires_on"])
+
 
 def cf_connectedk8s(cli_ctx, *_):
     from azext_connectedk8s.vendored_sdks import ConnectedKubernetesClient
+    if os.getenv('AZURE_ACCESS_TOKEN'):
```

Comment:
The name `AZURE_ACCESS_TOKEN` is never finalized. If https://github.com/Azure/azure-cli/issues/16459 is implemented in the future, it may change.

Reply:
This functionality is introduced for supporting internal partners only and hence we read the token via environment variables. If in future this gets implemented in az cli, we can use/expose the same parameters as defined by az cli